app/routers/event.py

```diff
@@ -516,6 +535,51 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
         return None
 
 
+def extract_shared_list_from_data(
+    event_info: ImmutableMultiDict,
+    db: Session,
+) -> Optional[SharedList]:
+    """Extract shared list items from POST data.
+    Return:
+        SharedList: SharedList object stored in the database.
+    """
+
+    items = zip(
+        event_info.getlist("item_name"),
+        event_info.getlist("item_amount"),
+        event_info.getlist("item_participant"),
+    )
+    shared_list = {"title": event_info.get("title"), "items": []}
+    for name, amount, participant in items:
+        item = SharedItem(name=name, amount=amount, participant=participant)
+        if _check_item_is_valid(item):
+            item_dict = item._asdict()
+            item_dict["amount"] = float(item_dict["amount"])
+            shared_list["items"].append(item_dict)
+    return _create_shared_list(shared_list, db)
+
+
+def _check_item_is_valid(item: SharedItem) -> bool:
+    return item != "" and item.amount.isnumeric() and item.participant != ""
```

Comment:
When item can be ""? maybe check if not None?

Reply:
Although the input elements of each item in the list are required to submit, this line check if an empty item submitted. If the field wasn't filled with anything, it will contain "".

Reply:
> If the field wasn't filled with anything, it will contain ""

You mean it was not filled in FE?
If you control that, maybe None it's better than empty string "" to represent an empty  SharedItem